website/client-old/js/services/questServices.js

```diff
@@ -82,33 +85,35 @@ angular.module('habitrpg')
           text += '**' + window.env.t('collect') + ':** ' + quest.collect[key].count + ' ' + quest.collect[key].text() + '\n\n';
         }
       }
-      text += '---\n\n';
-      text += '**' + window.env.t('rewardsAllParticipants') + ':**\n\n';
-      var participantRewards = _.reject(quest.drop.items, 'onlyOwner');
-      if(participantRewards.length > 0) {
-        _.each(participantRewards, function(item) {
-          text += item.text() + '\n\n';
-        });
-      }
-      if(quest.drop.exp)
-        text += quest.drop.exp + ' ' + window.env.t('experience') + '\n\n';
-      if(quest.drop.gp)
-        text += quest.drop.gp + ' ' + window.env.t('gold') + '\n\n';
-
-      var ownerRewards = _.filter(quest.drop.items, 'onlyOwner');
-      if(ownerRewards.length > 0) {
-        text += '**' + window.env.t('rewardsQuestOwner') + ':**\n\n';
-        _.each(ownerRewards, function(item){
-          text += item.text() + '\n\n';
-        });
+      if (quest.drop) {
```

Comment:
Previously, we assumed that if the quest data passed did not include `quest.boss`, then it must have `quest.drop`. This was not an extensible assumption--when I introduced bundles, which are neither bosses nor collection quests, this function would error out as soon as it tried to find `quest.drop.items`.